spec/index.bs

```diff
@@ -2344,6 +2689,51 @@ dictionary GPUVertexAttributeDescriptor {
 };
 </script>
 
+<dl dfn-type="abstract-op">
+    : <dfn>validating GPUVertexBufferLayoutDescriptor</dfn>(descriptor, vertexStage)
+    ::
+        <div algorithm="validation GPUVertexBufferLayoutDescriptor(descriptor, vertexStage)">
+            **Arguments:**
+                - {{GPUVertexBufferLayoutDescriptor}} |descriptor|
+                - {{GPUProgrammableStageDescriptor}} |vertexStage|
+
+            Return true, if and only if, all of the following conditions are true:
+
+            1. |descriptor|.{{GPUVertexBufferLayoutDescriptor/attributes}}.length is less than or equal to 16.
+            1. |descritor|.{{GPUVertexBufferLayoutDescriptor/arrayStride}} is less then or equal to 2048.
+            1. Any attribute |at| in the list |descriptor|.{{GPUVertexBufferLayoutDescriptor/attributes}} has
+                |at|.{{GPUVertexAttributeDescriptor/offset} + sizeOf(|at|.{{GPUVertexAttributeDescriptor/format}} less or equal to
+                |descritor|.{{GPUVertexBufferLayoutDescriptor/arrayStride}}.
+            1. For every vertex attribute in the shader reflection of |vertexStage|.{{GPUProgrammableStageDescriptor/module}}
+                that is know to be [=statically used=] by |vertexStage|.{{GPUProgrammableStageDescriptor/entryPoint}},
+                there is a corresponding |at| element of |descriptor|.{{GPUVertexBufferLayoutDescriptor/attributes}} that:
+                1. The shader format is |at|.{{GPUVertexAttributeDescriptor/format}}.
+                2. The shader location is |at|.{{GPUVertexAttributeDescriptor/shaderLocation}}.
+        </div>
+</dl>
+
+Issue(https://github.com/gpuweb/gpuweb/issues/693): add a limit to the number of vertex attributes
+
+<dl dfn-type="abstract-op">
+    : <dfn>validating GPUVertexStateDescriptor</dfn>(descriptor, vertexStage)
+    ::
+        <div algorithm="validation GPUVertexStateDescriptor(descriptor, vertexStage)">
```

Comment:
```html
<div algorithm>
    <dfn abstract-op>validating GPUVertexStateDescriptor</dfn>(descriptor, vertexStage)

    ...
</div>
```

Reply:
This works, thank you! Changes are applied now :)